src/ralph/networks/tests/test_models.py

```diff
@@ -197,6 +197,39 @@ def test_sub_network_should_assign_automatically(self):
         )
         self.assertEqual(net, subnet.parent)
 
+    def test_sub_network_should_change_automatically(self):
+        net1 = Network.objects.create(
+            name='net', address='10.20.30.0/24'
+        )
+        net2 = Network.objects.create(
+            name='net2', address='10.20.30.240/28'
+        )
+
+        self.assertEqual(net1, net2.parent)
+
+        net3 = Network.objects.create(
+            name='net3', address='10.20.30.128/25'
+        )
+
+        self.refresh_objects_from_db(net2)
+        self.assertEqual(net2.parent, net3)
+
+        self.assertEqual(net3.parent, net1)
+
+        net4 = Network.objects.create(
+            name='net4', address='10.20.40.128/28'
+        )
+        net3.address = '10.20.40.128/25'
+        net3.save()
+
+        self.refresh_objects_from_db(net1, net2, net3, net4)
+
+        self.assertEqual(net4.parent, net3)
+        self.assertEqual(net2.parent, net1)
+
+        net3.delete()
+        self.assertIsNone(Network.objects.filter(pk=net4.pk).first())
```

Comment:
is this desired behavior that deleting parent network will delete all subnetworks as well?